src/Behat/Testwork/ServiceContainer/ExtensionManager.php

```diff
@@ -34,30 +32,46 @@
     private $debugInformation = array(
         'extensions_list' => array()
     );
+    /**
+     * @var ExtensionInstantiator[]
+     */
+    private $instantiators = array();
 
     /**
      * Initializes manager.
      *
-     * @param Extension[] $extensions     List of default extensions
-     * @param null|string $extensionsPath Base path where to search custom extension files
+     * @param Extension[]             $extensions      List of default extensions
+     * @param string|null             $extensionsPath  Base path where to search custom extension files (deprecated)
+     * @param ExtensionInstantiator[] $instantiators   Extension instantiators to use
      */
-    public function __construct(array $extensions, $extensionsPath = null)
+    public function __construct(array $extensions, $extensionsPath = null, array $instantiators = array())
```

Comment:
the `extensionsPath` argument should be deprecated.

So instantiators should be passed as second argument in the non-deprecated signature. The BC layer would enter into action when it is not an array.

your current logic would force keeping the `$extensionsPath` argument in 4.0 when removing the BC layer.